omega/omega-transaction/src/main/java/org/apache/servicecomb/pack/omega/transaction/TransactionAspect.java

```diff
@@ -46,6 +48,15 @@ public TransactionAspect(SagaMessageSender sender, OmegaContext context) {
   @Around("execution(@org.apache.servicecomb.pack.omega.transaction.annotations.Compensable * *(..)) && @annotation(compensable)")
   Object advise(ProceedingJoinPoint joinPoint, Compensable compensable) throws Throwable {
     Method method = ((MethodSignature) joinPoint.getSignature()).getMethod();
+    // just check if we need to setup the transaction context information first
+    TransactionContext transactionContext = getTransactionContextFromArgs(joinPoint.getArgs());
+    if (transactionContext != null) {
+      LOG.debug("Updated context {} for globalTxId:{} and localTxId:{}", context,
+          transactionContext.globalTxId(), transactionContext.localTxId());
+      context.setGlobalTxId(transactionContext.globalTxId());
+      context.setLocalTxId(transactionContext.localTxId());
+    }
```

Comment:
It could be better to restore the origin globalTxId and localTxId if they exists.

Reply:
As we just get the transactionContext from business object, we need to pass them into next invocation as the omega transport does. Any thought？

Reply:
So in this situation, the origin globalTxId and localTxId should be empty here. Can you log some warning message if the globalTxId is not empty ?

Reply:
I agree，we need to warning the user about it.